src/System.Device.Gpio.Tests/README.md

```diff
@@ -1,52 +1,127 @@
 - Leave BCM Pin 23 (physical Pin 16) open (not connected to anything).
 
 ## Raspberry Pi driver tests
+
 After that, you can run the tests with the RaspberryPiDriver (which is the default low-level driver for the Raspberry Pi) like:
+
 ```
 dotnet test --filter RaspberryPiDriverTests System.Device.Gpio.sln 
 ```
+
 Depending on the version of the Pi and the installed Linux distribution, it may be required to run the tests as root:
+
 ```
 sudo dotnet test --filter RaspberryPiDriverTests System.Device.Gpio.sln 
 ```
 
 If everything went smoothly, the output should end with a success message. 
 
+## Running tests depending on components/requirements
+
+Currently the full test suite requires following components, if you don't have one use following command line switches to skip
+
+| Requirements | Skip arguments |
+| --- | --- |
+| MCP 3008 (SPI)* | `-notrait feature=spi` |
+| BME280 (I2C) | `-notrait feature=i2c` |
+| Pins 5 and 6 connected | `-notrait feature=gpio` |
+| libgpiod | `-notrait feature=gpio-libgpiod` |
+| sysfs access (i.e. sudo or permissions) | `-notrait feature=gpio-sysfs` |
+| configured PWM (overlaps with MCP3008 setting) | `-notrait feature=pwm` |
+| root access (overlaps, assumes you use default permissions) | `-notrait requirement=root` |
+
+* Also inputs to ADC are required including one connected to PWM output through low pass filter, please refer to the schematics.
+
+
+
 ## SysFsDriver Tests
+
 You can also run the SysFsDriver Tests (this uses a more generic approach). This driver requires root permissions to run, so you need to `sudo` the command:
+
 ```
 sudo dotnet test --filter SysFsDriverTests System.Device.Gpio.sln 
 ```
 
 ## LibgpiodDriver Tests
+
 To run the Libgpiod Driver test (a Linux Kernel Driver for the GPIO device), you need to first install the libgpiod package:
+
 ```
 sudo apt install -y libgpiod-dev
 ```
+
 These tests do not require root permissions to run:
+
 ```
 dotnet test --filter LibGpiodDriverTests System.Device.Gpio.sln 
 ```
+
+## Full test run schematics
+
+Our CI is currently using [this schematics](schematics/board-schematics.pdf) with Raspberry PI 3.
+
+You can download [gerber files](schematics/board-gerber.zip) if you'd like to manufacture PCB yourself (this is a 2 layer board).
+
+Following list of components are needed for PCB:
+
+| Component | Quantity |
+| --- | --- |
+| MCP3008 I/SL | 1 |
+| BME280 breakout board | 1 |
+| 2x20 female header (Raspberry PI hat) | 1 |
+| 1x7 female header (for BME280 breakout board) | 1 |
+| 4.7k resistor in 0603 (US) package | 4 |
+| 0.1u (100n) capacitor in 0603 (US) package | 1 |
+| (optional) 1x3 male header | 5 |
+| (optional) 1x4 male header | 4 |
+| (optional) 1x8 male header | 1 |
+| (optional) 200 ohm resistor in 0603 (US) package | 1 |
+| (optional) LED in 0603 (US) package | 1 |
```

Comment:
Minor: If connecting an LED an additional resistor (~150 Ohms) is required, or this will be a short-living experience...

Reply:
I mentioned this line above: `| (optional) 200 ohm resistor in 0603 (US) package | 1 |` - I usually use 200 ohms for LEDs

Reply:
Uh sorry, I have overlooked that line.